packages/inventory-insights/src/Insights.js

```diff
@@ -275,8 +266,40 @@ class InventoryRuleList extends Component {
         this.setState({ searchValue: value, rows });
     };
 
+    onSelect = (event, isSelected, rowId) => {
+        let rows;
+        if (rowId === -1) {
+            rows = this.state.rows.map(oneRow => {
+                oneRow.selected = isSelected;
+                return oneRow;
+            });
+        } else {
+            rows = [ ...this.state.rows ];
+            rows[rowId].selected = isSelected;
+        }
+
+        this.setState({
+            rows
```

Comment:
Spread might be better:
```JS
onSelect = (event, isSelected, rowId) => {
  this.setState({
    rows: this.state.rows.map((oneRow, rowKey) => {
        if (rowId === -1 || rowKey === rowId) {
           oneRow.selected = isSelected;
        }
        return row;
      })
  })
}
```

Reply:
yeah! we can do this, but probs want to return `oneRow` rather than row 👍

Reply:
we could even go further... turn it into a one liner `             rows: this.state.rows .map((oneRow, rowKey) => rowId === -1 || rowKey === rowId ? { ...oneRow, selected: isSelected } : { ...oneRow })`